ietf-dtnma-agent.yang

```diff
@@ -641,12 +657,131 @@ module ietf-dtnma-agent {
       amm:column max-count {
         amm:type "/ARITYPE/UVAST";
       }
+      amm:column init-enabled {
+        amm:type "/ARITYPE/BOOL";
+      }
       amm:column enabled {
         amm:type "/ARITYPE/BOOL";
       }
     }
   }
 
+  // CTRLs related to rules
+  amm:ctrl ensure-sbr {
+    if-feature "rules";
+    amm:enum 13;
+    description
+      "Ensure SBR with the object-id exists with the given parameters.";
+    uses object-id;
+    amm:parameter action {
+      description
+        "The execution when this rule triggers.";
+      amm:type "//ietf/amm/TYPEDEF/MAC";
+    }
+    amm:parameter start-time {
+      amm:type "//ietf/amm/TYPEDEF/TIME";
+    }
+    amm:parameter condition {
+      amm:type "//ietf/amm/TYPEDEF/EXPR";
+    }
+    amm:parameter min-interval {
+      amm:type "/ARITYPE/TD";
+    }
+    amm:parameter max-count {
+      amm:type "/ARITYPE/UVAST";
+    }
+    amm:parameter init-enabled {
+      amm:type "/ARITYPE/BOOL";
+    }
+    amm:result res {
+      amm:type "/ARITYPE/UINT";
+    }
+  }
+  amm:ctrl ensure-tbr {
+    if-feature "rules";
+    amm:enum 14;
+    description
+      "Ensure TBR with the object-id exists with the given parameters.";
+    uses object-id;
+    amm:parameter action {
+      description
+        "The execution when this rule triggers.";
+      amm:type "//ietf/amm/TYPEDEF/MAC";
+    }
+    amm:parameter start-time {
+      amm:type "//ietf/amm/TYPEDEF/TIME";
+    }
+    amm:parameter period {
+      amm:type "/ARITYPE/TD";
+    }
+    amm:parameter max-count {
+      amm:type "/ARITYPE/UVAST";
+    }
+    amm:parameter max-count {
+      amm:type "/ARITYPE/UVAST";
+    }
+    amm:parameter init-enabled {
+      amm:type "/ARITYPE/BOOL";
+    }
+    amm:result res {
+      amm:type "/ARITYPE/UINT";
+    }
+  }
+  amm:ctrl ensure-enabled {
+    if-feature "rules";
+    amm:enum 15;
+    description
+      "Ensure rule is enabled or disabled.";
+    amm:parameter obj-id {
```

Comment:
I assume there would be no restrictions on ensuring or resetting enabled status for ADM objects.